staging/src/k8s.io/apiserver/pkg/server/storage/storage_factory.go

```diff
@@ -252,15 +262,23 @@ func (s *DefaultStorageFactory) NewConfig(groupResource schema.GroupResource) (*
 	return &storageConfig, nil
 }
 
-// Get all backends for all registered storage destinations.
+// Backends returns all backends for all registered storage destinations.
 // Used for getting all instances for health validations.
-func (s *DefaultStorageFactory) Backends() []string {
-	backends := sets.NewString(s.StorageConfig.ServerList...)
+func (s *DefaultStorageFactory) Backends() []Backend {
+	servers := sets.NewString(s.StorageConfig.ServerList...)
 
```

Comment:
if `Backend{}` takes a *tls.Config, you can also `--etcd-cafile=...` if passed, you only need to parse `--etcd-certfile=... --etcd-keyfile=...` once, and inner layers don't need to be file-aware